rfcs/text/0017_url_service.md

```diff
@@ -0,0 +1,462 @@
+  },
+});
+```
+
+These short URL will be stored in saved objects of type `url` and will be
+automatically migrated once a URL generator becomes deprecated.
+
+
+## HTTP endpoints
+
+
+### Short URL CRUD+ HTTP endpoints
+
+Below HTTP endpoints are designed to work specifically with short URLs:
+
+| HTTP method           | Path                                | Description                                                                                                                              |
+|-----------------------|-------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| __POST__              | `/api/short_url`                    | Endpoint for creating new short URLs.                                                                                                    |
+| __GET__               | `/api/short_url/<slug>`             | Endpoint for retrieving information about an existing short URL.                                                                         |
+| __DELETE__            | `/api/short_url/<slug>`             | Endpoint for deleting an existing short URL.                                                                                             |
+| __POST__              | `/api/short_url/<slug>`             | Endpoint for updating information about an existing short URL.                                                                           |
+| __GET__ or __POST__   | `/api/short_url/<slug>/_resolve`    | Same as `GET /api/short_url/<slug>`, but this endpoint also increments the short URL access count counter and the last access timestamp. |
+| __GET__ or __POST__   | `/api/short_url/<slug>/_redirect`   | Redirects the user using HTTP 301 (permanent) redirect to the destination long URL.                                                      |
+
+
+
+### Other URL Service HTTP endpoints
+
+Below are all the other HTTP endpoints that URL Service will provide:
+
+| HTTP method           | Path                                | Description                                                                                                                              |
+|-----------------------|-------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| __GET__ or __POST__   | `/api/url/_redirect`                | Redirects the user using HTTP 302 (temporary) redirect. Receives URL generator ID and parameters in the POST body payload.               |
+| __GET__ or __POST__   | `/go/<slug>`                        | A very short alias for `POST /api/share/url/<slug>/_redirect` endpoint.                                                                  |
+
+
+#### The URL generator "redirect" endpoint
+
+The `/api/url/_redirect` endpoint will receive the necessary payload to
```

Comment:
does this exist on client/server contracts as well ?

Reply:
Updated RFC to only have it on client. But there will still be a legacy endoint on server, from old short URL service.